var/spack/repos/builtin/packages/cmake/package.py

```diff
@@ -166,12 +177,12 @@ class Cmake(Package):
     depends_on('libuv@1.10.0:1.10.99',  when='@3.11.0:3.11.99~ownlibs')
     depends_on('libuv@1.10.0:',  when='@3.12.0:~ownlibs')
     depends_on('rhash',          when='@3.8.0:~ownlibs')
-    depends_on('qt',             when='+qt')
-    depends_on('python@2.7.11:', when='+doc', type='build')
-    depends_on('py-sphinx',      when='+doc', type='build')
-    depends_on('openssl', when='+openssl')
-    depends_on('openssl@:1.0.99', when='@:3.6.9+openssl')
-    depends_on('ncurses',        when='+ncurses')
+    depends_on('qt',             when='+qt~ownlibs')
+    depends_on('python@2.7.11:', when='+doc~ownlibs', type='build')
+    depends_on('py-sphinx',      when='+doc~ownlibs', type='build')
+    depends_on('openssl', when='+openssl~ownlibs')
```

Comment:
```suggestion
    depends_on('openssl', when='+ownlibs')
```
I think this `openssl` variant was generally flawed, see #25559, but I realize now that my PR #25559 was wrong. What we need is that cmake bootstraps curl, and curl needs an external/system openssl. So effectively we really shouldn't have an `openssl` variant at all, it's a requirement when bootstrapping curl.